### PR TITLE
chore(swap): Use CHAINFLIP_NETWORK env variable to determine appropriate address validation

### PR DESCRIPTION
**Pull request by niklasnatter**

### CODE REVIEW

File: packages/shared/src/validation/addressValidation.ts [hunk 1]
```diff
@@ -108,25 +108,32 @@ export const validateBitcoinRegtestAddress: AddressValidator = (
...
-  isMainnet = true,
+  network: ChainflipNetwork,
 ): boolean => {
   switch (chain) {
     case 'Bitcoin':
-      return isMainnet
-        ? validateBitcoinMainnetAddress(address)
-        : validateBitcoinTestnetAddress(address) ||
-            validateBitcoinRegtestAddress(address);
+      switch (network) {
```
Comment (acdibble) [inline packages/shared/src/validation/addressValidation.ts:115]: looks like an opportunity for another map: `Record<ChainflipNetwork, Record<ChainflipChain, (addr: string) => boolean>>`

File: packages/swap/src/handlers/openSwapDepositChannel.ts [hunk 2]
```diff
@@ -12,7 +12,13 @@ import ServiceError from '../utils/ServiceError';
 export default async function openSwapDepositChannel(
   input: z.output<typeof openSwapDepositChannelSchema>,
 ) {
-  if (!validateAddress(input.destChain, input.destAddress, isProduction)) {
+  if (
+    !validateAddress(
+      input.destChain,
+      input.destAddress,
+      process.env.CHAINFLIP_NETWORK as ChainflipNetwork,
```
Comment (acdibble) [inline packages/swap/src/handlers/openSwapDepositChannel.ts:19]: could do an zod env migration here

**Review by acdibble — APPROVED** [review]
was also thinking about this last night

File: packages/swap/src/handlers/openSwapDepositChannel.ts (hunk 2 above)
Comment (niklasnatter) [reply packages/swap/src/handlers/openSwapDepositChannel.ts:19]: yes this should be the next app we tackle 🙂

**Review by GabrielBuragev — APPROVED** [review]

File: packages/shared/src/validation/addressValidation.ts (hunk 1 above)
Comment (niklasnatter) [reply packages/shared/src/validation/addressValidation.ts:115]: it looks like that indeed, but im too lazey to do right now 😇

**Pull request merged**